tests/test_async_with.py

```diff
@@ -131,7 +131,7 @@ async def test_create_pool_deprecations(mysql_params, loop):
             warnings.simplefilter("always")
             async with pool.get() as conn:
                 pass
-    assert issubclass(w[-1].category, DeprecationWarning)
+    assert issubclass(w[0].category, DeprecationWarning)
```

Comment:
Why is this important?

Reply:
This is necessary for tests to pass on P3.7 with `PYTHONASYNCIODEBUG=1`, for which there are two warnings emitted:
```
(Pdb) print([x.category for x in w])
[<class 'DeprecationWarning'>, <class 'ResourceWarning'>]
```
It doesn't happen on P3.8 though. Here's related build result:
https://travis-ci.com/github/aio-libs/aiomysql/jobs/341264313

Reply:
The ResourceWarning msg:
```
ResourceWarning('unclosed event loop <_UnixSelectorEventLoop running=False closed=False debug=True>')
```
coming from `asyncio/base_events.py` line 623.

Reply:
Okay, in this case, we should add a code comment explaining this magic of choosing which element to compare.

Reply:
I have added a code comment. I have also added some explanation in the commit msg.

Reply:
Honestly that `ResourceWarning` is a bug that may mean that our tests don't close the loop properly, upon teardown. Does this happen under Python 3.8?

Reply:
No, this happens only under 3.7